Python/instrumentation.c

```diff
@@ -1702,10 +1702,12 @@ instrument_lock_held(PyCodeObject *code, PyInterpreterState *interp)
         );
         return 0;
     }
+#ifdef _Py_TIER2
```

Comment:
Reading through this, I'm constantly torn between doing what you've done here or just making all of these APIs no-ops when tier two isn't built. I think the way you've done it is probably better, but all of these `#ifdef`s make me a tiny bit sad.

Reply:
Totally. What I did is easier, because otherwise I'd have to selectively keep certain parts of optimizer.c where the functionality is implemented. That would be a mess of functions surrounded by `#endif ... #ifdef`, or I'd have to move everything to the end of the file. I tend to prefer the shortest diff.